src/main/java/com/faforever/api/user/UsersController.java

```diff
@@ -51,86 +52,136 @@ public void register(HttpServletRequest request,
+  @PostMapping(path = "/requestPasswordReset", produces = APPLICATION_JSON_VALUE)
   public void requestPasswordReset(@RequestParam("identifier") String identifier) {
     userService.requestPasswordReset(identifier);
   }
 
   @ApiOperation("Sets a new password for an account.")
-  @RequestMapping(path = "/performPasswordReset", method = RequestMethod.POST, produces = APPLICATION_JSON_VALUE)
+  @PostMapping(path = "/performPasswordReset", produces = APPLICATION_JSON_VALUE)
   public void performPasswordReset(HttpServletResponse response,
                                    @RequestParam("token") String token,
-                                   @RequestParam("newPassword") String newPassword) throws IOException {
+                                   @RequestParam("newPassword") String newPassword) {
     userService.performPasswordReset(token, newPassword);
   }
 
   @PreAuthorize("#oauth2.hasScope('" + OAuthScope._WRITE_ACCOUNT_DATA + "') and hasRole('ROLE_USER')")
   @ApiOperation("Creates an URL to the steam platform to initiate the Link To Steam process.")
-  @RequestMapping(path = "/buildSteamLinkUrl", method = RequestMethod.POST, produces = APPLICATION_JSON_VALUE)
-  public Map<String, Serializable> buildSteamLinkUrl(Authentication authentication, @RequestParam("callbackUrl") String callbackUrl) {
+  @PostMapping(path = "/buildSteamLinkUrl", produces = APPLICATION_JSON_VALUE)
+  public Map<String, Serializable> buildSteamLinkUrl(Authentication authentication,
+                                                     @RequestParam("callbackUrl") String callbackUrl) {
     String steamUrl = userService.buildSteamLinkUrl(userService.getUser(authentication), callbackUrl);
     return Map.of("steamUrl", steamUrl);
   }
 
+  @ApiOperation("Build a password reset link via Steam.")
+  @PostMapping(path = "/buildSteamPasswordResetUrl", produces = APPLICATION_JSON_VALUE)
+  public Map<String, Serializable> buildSteamPasswordResetUrl() {
+    String steamUrl = userService.buildSteamPasswordResetUrl();
+    return Map.of("steamUrl", steamUrl);
+  }
+
+  @ApiOperation("Sends a password reset request to the username OR email linked by this account.")
+  @GetMapping(path = "/requestPasswordResetViaSteam", produces = APPLICATION_JSON_VALUE)
+  public void requestPasswordResetViaSteam(HttpServletRequest request,
+                                           HttpServletResponse response) {
+    steamService.validateSteamRedirect(request);
+    String steamId = steamService.parseSteamIdFromLoginRedirect(request);
+    CallbackResult result = userService.requestPasswordResetViaSteam(steamId);
+    redirectCallbackResult(response, result);
+  }
+
+  @ApiOperation("Sets a new password for an account via Steam redirect.")
+  @GetMapping(path = "/performPasswordResetViaSteam", produces = APPLICATION_JSON_VALUE)
+  public void performPasswordResetViaSteam(HttpServletRequest request,
```

Comment:
also this is not used right? both password resets end in the same endpoint

Reply:
Yes, the new approach is much leaner.